docs/end_to_end_tutorials/chatbots/building_a_chatbot.md

```diff
@@ -225,129 +165,188 @@ We can now test the agent with various queries.
+
+2022 Risk Factors:
+- Potential adverse effect if drivers were classified as employees instead of independent contractors.
+- Highly competitive nature of the mobility, delivery, and logistics industries.
+- Need to lower fares or service fees to remain competitive.
+- History of significant losses and expectation of increased operating expenses.
+- Impact of future pandemics or disease outbreaks on the business and financial results.
+- Potential harm to the business due to economic conditions and their effect on discretionary consumer spending.
+
+2021 Risk Factors:
+- Adverse impact of the COVID-19 pandemic and actions to mitigate it on the business.
+- Potential reclassification of drivers as employees instead of independent contractors.
+- Highly competitive nature of the mobility, delivery, and logistics industries.
+- Need to lower fares or service fees and offer incentives to remain competitive.
+- History of significant losses and uncertainty of achieving profitability.
+- Importance of attracting and maintaining a critical mass of platform users.
+
+2020 Risk Factors:
+- Adverse impact of the COVID-19 pandemic on the business.
+- Potential reclassification of drivers as employees.
+- Highly competitive nature of the mobility, delivery, and logistics industries.
+- Need to lower fares or service fees to remain competitive.
+- History of significant losses and potential future expenses.
+- Importance of attracting and maintaining a critical mass of platform users.
+- Operational and cultural challenges faced by the company.
+
+2019 Risk Factors:
+- Competition with local companies.
+- Differing levels of social acceptance.
+- Technological compatibility issues.
+- Exposure to improper business practices.
+- Legal uncertainty.
+- Difficulties in managing international operations.
+- Fluctuations in currency exchange rates.
+- Regulations governing local currencies.
+- Tax consequences.
+- Financial accounting burdens.
+- Difficulties in implementing financial systems.
+- Import and export restrictions.
+- Political and economic instability.
+- Public health concerns.
+- Reduced protection for intellectual property rights.
+- Limited influence over minority-owned affiliates.
+- Regulatory complexities.
+
+These comparisons highlight both common and unique risk factors that Uber faced in different years.
+```
 
 ### Setting up the Chatbot Loop
 
-Now that we have the chatbot setup, it only takes a few more steps to setup a basic interactive loop to converse with our SEC-augmented chatbot! 
+Now that we have the chatbot setup, it only takes a few more steps to setup a basic interactive loop to chat with our SEC-augmented chatbot!
 
 ```python
+agent = OpenAIAgent.from_tools(tools) # verbose=False by default
+
 while True:
     text_input = input("User: ")
-    response = agent_chain.run(input=text_input)
-    print(f'Agent: {response}')
-    
+    response = agent.chat(text_input)
+    print(f"Agent: {response}")
```

Comment:
Can we add a quick keyword check for exiting the loop? It will help the notebook since there won't be a huge stack trace when you quit